From 340cdd3c9648c4b35ef95cc0498413b763e71859 Mon Sep 17 00:00:00 2001
From: William Brown <firstyear@redhat.com>
Date: Sun, 31 Jan 2016 14:27:13 +1000
Subject: [PATCH] Ticket 48448 - dirsrv start-stop fail in certain shell
 environments.

Bug Description:  Dirsrv fails to start and stop with certain shell environments
. This is due to the usage of the pattern

  cd $SERVERBIN_DIR
  ./ns-slapd ...

Fix Description:  Change all invocations of commands to use explicit paths:

  $SERVERBIN_DIR/ns-slapd

https://fedorahosted.org/389/ticket/48448

Author: wibrown

Review by: ???
---
 ldap/admin/src/scripts/start-dirsrv.in             | 2 +-
 ldap/admin/src/scripts/template-bak2db.in          | 3 +--
 ldap/admin/src/scripts/template-db2bak.in          | 5 ++---
 ldap/admin/src/scripts/template-db2index.in        | 5 ++---
 ldap/admin/src/scripts/template-db2ldif.in         | 3 +--
 ldap/admin/src/scripts/template-dbverify.in        | 5 ++---
 ldap/admin/src/scripts/template-dn2rdn.in          | 5 ++---
 ldap/admin/src/scripts/template-ldif2db.in         | 3 +--
 ldap/admin/src/scripts/template-ldif2ldap.in       | 5 ++---
 ldap/admin/src/scripts/template-monitor.in         | 5 ++---
 ldap/admin/src/scripts/template-restoreconfig.in   | 5 ++---
 ldap/admin/src/scripts/template-saveconfig.in      | 5 ++---
 ldap/admin/src/scripts/template-suffix2instance.in | 5 ++---
 ldap/admin/src/scripts/template-upgradedb.in       | 3 +--
 ldap/admin/src/scripts/template-upgradednformat.in | 5 ++---
 ldap/admin/src/scripts/template-vlvindex.in        | 3 +--
 16 files changed, 26 insertions(+), 41 deletions(-)

diff --git a/ldap/admin/src/scripts/start-dirsrv.in b/ldap/admin/src/scripts/start-dirsrv.in
index 458f0e8..513addb 100755
--- a/ldap/admin/src/scripts/start-dirsrv.in
+++ b/ldap/admin/src/scripts/start-dirsrv.in
@@ -70,7 +70,7 @@ start_instance() {
             return 1
         fi
     else
-        cd $SERVERBIN_DIR; ./ns-slapd -D $CONFIG_DIR -i $PIDFILE -w $STARTPIDFILE "$@"
+        $SERVERBIN_DIR/ns-slapd -D $CONFIG_DIR -i $PIDFILE -w $STARTPIDFILE "$@"
         if [ $? -ne 0 ]; then
             return 1
         fi
diff --git a/ldap/admin/src/scripts/template-bak2db.in b/ldap/admin/src/scripts/template-bak2db.in
index 1725aba..01a12fa 100755
--- a/ldap/admin/src/scripts/template-bak2db.in
+++ b/ldap/admin/src/scripts/template-bak2db.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./bak2db "$@" -Z {{SERV-ID}}
+{{SERVERBIN-DIR}}/bak2db "$@" -Z {{SERV-ID}}
 exit $?
diff --git a/ldap/admin/src/scripts/template-db2bak.in b/ldap/admin/src/scripts/template-db2bak.in
index 50fd5ed..70a4a2b 100755
--- a/ldap/admin/src/scripts/template-db2bak.in
+++ b/ldap/admin/src/scripts/template-db2bak.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./db2bak "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/db2bak "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-db2index.in b/ldap/admin/src/scripts/template-db2index.in
index a7a633d..9c7c5ec 100755
--- a/ldap/admin/src/scripts/template-db2index.in
+++ b/ldap/admin/src/scripts/template-db2index.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./db2index "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/db2index "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-db2ldif.in b/ldap/admin/src/scripts/template-db2ldif.in
index b85ffdc..3881911 100755
--- a/ldap/admin/src/scripts/template-db2ldif.in
+++ b/ldap/admin/src/scripts/template-db2ldif.in
@@ -1,6 +1,5 @@
 #!/bin/sh
 
 cwd=`pwd`
-cd {{SERVERBIN-DIR}}
-./db2ldif "$@" -Z {{SERV-ID}} -c $cwd
+{{SERVERBIN-DIR}}/db2ldif "$@" -Z {{SERV-ID}} -c $cwd
 exit $?
diff --git a/ldap/admin/src/scripts/template-dbverify.in b/ldap/admin/src/scripts/template-dbverify.in
index 71e3e4e..abcc58e 100755
--- a/ldap/admin/src/scripts/template-dbverify.in
+++ b/ldap/admin/src/scripts/template-dbverify.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./dbverify "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/dbverify "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-dn2rdn.in b/ldap/admin/src/scripts/template-dn2rdn.in
index b3d8e82..9ecae08 100755
--- a/ldap/admin/src/scripts/template-dn2rdn.in
+++ b/ldap/admin/src/scripts/template-dn2rdn.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./dn2rdn "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/dn2rdn "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-ldif2db.in b/ldap/admin/src/scripts/template-ldif2db.in
index f3fa58e..f38fce3 100755
--- a/ldap/admin/src/scripts/template-ldif2db.in
+++ b/ldap/admin/src/scripts/template-ldif2db.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./ldif2db "$@" -Z {{SERV-ID}}
+{{SERVERBIN-DIR}}/ldif2db "$@" -Z {{SERV-ID}}
 exit $?
diff --git a/ldap/admin/src/scripts/template-ldif2ldap.in b/ldap/admin/src/scripts/template-ldif2ldap.in
index 806ddcc..c785742 100755
--- a/ldap/admin/src/scripts/template-ldif2ldap.in
+++ b/ldap/admin/src/scripts/template-ldif2ldap.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./ldif2ldap "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/ldif2ldap "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-monitor.in b/ldap/admin/src/scripts/template-monitor.in
index 2f93337..c89bb8a 100755
--- a/ldap/admin/src/scripts/template-monitor.in
+++ b/ldap/admin/src/scripts/template-monitor.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./monitor "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/monitor "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-restoreconfig.in b/ldap/admin/src/scripts/template-restoreconfig.in
index f4b2d06..5109561 100755
--- a/ldap/admin/src/scripts/template-restoreconfig.in
+++ b/ldap/admin/src/scripts/template-restoreconfig.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./restoreconfig "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/restoreconfig "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-saveconfig.in b/ldap/admin/src/scripts/template-saveconfig.in
index c77cce1..7784e83 100755
--- a/ldap/admin/src/scripts/template-saveconfig.in
+++ b/ldap/admin/src/scripts/template-saveconfig.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./saveconfig "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/saveconfig "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-suffix2instance.in b/ldap/admin/src/scripts/template-suffix2instance.in
index 03bcba8..e29408d 100755
--- a/ldap/admin/src/scripts/template-suffix2instance.in
+++ b/ldap/admin/src/scripts/template-suffix2instance.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./suffix2instance "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/suffix2instance "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-upgradedb.in b/ldap/admin/src/scripts/template-upgradedb.in
index ce879bb..ae28ac2 100755
--- a/ldap/admin/src/scripts/template-upgradedb.in
+++ b/ldap/admin/src/scripts/template-upgradedb.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./upgradedb "$@" -Z {{SERV-ID}}
+{{SERVERBIN-DIR}}/upgradedb "$@" -Z {{SERV-ID}}
 exit $?
diff --git a/ldap/admin/src/scripts/template-upgradednformat.in b/ldap/admin/src/scripts/template-upgradednformat.in
index 5fd8ef9..74c18e8 100755
--- a/ldap/admin/src/scripts/template-upgradednformat.in
+++ b/ldap/admin/src/scripts/template-upgradednformat.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./upgradednformat "$@" -Z {{SERV-ID}}
-exit $?
\ No newline at end of file
+{{SERVERBIN-DIR}}/upgradednformat "$@" -Z {{SERV-ID}}
+exit $?
diff --git a/ldap/admin/src/scripts/template-vlvindex.in b/ldap/admin/src/scripts/template-vlvindex.in
index 0249696..a7ffb40 100755
--- a/ldap/admin/src/scripts/template-vlvindex.in
+++ b/ldap/admin/src/scripts/template-vlvindex.in
@@ -1,5 +1,4 @@
 #!/bin/sh
 
-cd {{SERVERBIN-DIR}}
-./vlvindex "$@" -Z {{SERV-ID}}
+{{SERVERBIN-DIR}}/vlvindex "$@" -Z {{SERV-ID}}
 exit $?
-- 
2.5.0

